libsnark/zk_proof_systems/plonk/utils.tcc

```diff
@@ -297,6 +297,24 @@ std::vector<FieldT> plonk_compute_accumulator(
     return A;
 }
 
+template<typename FieldT>
+std::vector<std::vector<FieldT>> plonk_gates_matrix_transpose(
+    const std::vector<std::vector<FieldT>> &gates_matrix,
+    const size_t &nrows,
+    const size_t &ncols)
+{
+    size_t nrows_transpose = ncols;
```

Comment:
check that `gates_matrix` has the expected size (at least num rows for now, and then possible the size fo each row)?  Or perhaps `ncols` and `nrows` can be extracted from `gates_matrix`, and we check that each row has the same length as the first?
Or are the rows *at most* `ncols`?

In either case, I think there are some asserts that could be helpful here.

Reply:
> Or perhaps ncols and nrows can be extracted from gates_matrix, and we check that each row has the same length as the first?

Applied the above suggestion.